File: core/src/main/java/cz/dynawest/jtexy/util/Debug.java

package cz.dynawest.jtexy.util;

import cz.dynawest.jtexy.ContentType;

/**
 *
 * @author Ondrej Zizka
 */
public class Debug {

    public static String showCodes( String str ){
        return str.replace(ContentType.BLOCK   .getDelim(), '▉')
                 .replace(ContentType.TEXTUAL .getDelim(), '₮')
                 .replace(ContentType.MARKUP  .getDelim(), 'ℳ')
                 .replace(ContentType.REPLACED.getDelim(), '℞');
    }
    
    public static String showCodesWithNumbers( String str ){
        str = str.replace(ContentType.BLOCK   .getDelim(), '▉')
                 .replace(ContentType.TEXTUAL .getDelim(), '₮')
                 .replace(ContentType.MARKUP  .getDelim(), 'ℳ')
                 .replace(ContentType.REPLACED.getDelim(), '℞');
        
        // TODO: Show numbers inside delimiters.
        
        return str;
    }
    
}// class

File: core/src/main/java/cz/dynawest/jtexy/util/Debug.kt
package cz.dynawest.jtexy.util

import cz.dynawest.jtexy.ContentType

/**
 *
 * @author Ondrej Zizka
 */
object Debug {
    fun showCodes(str: String?): String {
        return str!!.replace(ContentType.BLOCK.delim, '▉')
            .replace(ContentType.TEXTUAL.delim, '₮')
            .replace(ContentType.MARKUP.delim, 'ℳ')
            .replace(ContentType.REPLACED.delim, '℞')
    }

    fun showCodesWithNumbers(str: String): String {
        var str = str
        str = str.replace(ContentType.BLOCK.delim, '▉')
            .replace(ContentType.TEXTUAL.delim, '₮')
            .replace(ContentType.MARKUP.delim, 'ℳ')
            .replace(ContentType.REPLACED.delim, '℞')

        // TODO: Show numbers inside delimiters.
        return str
    }
}
